src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -103,10 +103,13 @@ internal class AnalysisCache
                 var moduleManifestProperties = PsUtils.GetModuleManifestProperties(modulePath, PsUtils.FastModuleManifestAnalysisPropertyNames);
                 if (moduleManifestProperties != null)
                 {
-                    if (ModuleIsEditionIncompatible(modulePath, moduleManifestProperties))
+                    if (!ExperimentalFeature.IsEnabled("PSWinCompat"))
```

Comment:
Merged into single `if`.
About caching...    I see your point. This would be a valid perf optimization if `ExperimentalFeature.IsEnabled` was expensive perf-wise, but it is not - perf gain in caching it's result is negligible. The negative side of caching is that it exposes internal behavior of `ExpFeatures` component (the fact that currently it does not change after start) to the outside (in this case - modules code and other places); so if behavior changes caching would break things.  The interface of `ExpFeatures` component is `ExperimentalFeature.IsEnabled`; if it is not fast enough (which it currently is) then it should be optimized internally without breaking the interface.